cub/test/catch2_test_triple_chevron.cu

```diff
@@ -0,0 +1,211 @@
+    for(int i = 0; i < std::min(5, n); ++i) {
+        printf("h_out[%d] = %d\n", i, h_out[i]);
+    }
+    const auto actual   = static_cast<std::size_t>(thrust::count(out.begin(), out.end(), 42));
+    printf("Thrust count result: %zu\n", actual);
+    const auto expected = static_cast<std::size_t>(n);
+
+    REQUIRE(actual == expected);
+  }
+
+
+}
+
+
+CUB_TEST("Rough draft of testing Chevron launches successfully ", "[test][utils]") {
+  cudaStream_t stream1, stream2;
+  cudaStreamCreate(&stream1);
+  cudaStreamCreate(&stream2);
+  
+  cudaEvent_t event;
+  cudaEventCreate(&event);
+
+  int n = 1000;
+  thrust::device_vector<int> in(n, 5);
+  thrust::device_vector<int> out(n);
+
+  int *d_in  = thrust::raw_pointer_cast(in.data());
+  int *d_out = thrust::raw_pointer_cast(out.data());
+
+  cudaMemcpyAsync(d_out, d_in, n * sizeof(float), cudaMemcpyHostToDevice, stream1); 
+  cudaEventRecord(event, stream1); 
+
+  const int block_size = 256;
+  const int grid_size = (n * block_size - 1) / block_size;
+  auto chev = cub::detail::triple_chevron(grid_size, block_size, 0, stream2);
+
+  cudaStreamWaitEvent(stream2, event, 0); // sync streams
+  chev.doit(mult_two_kernel<int>, d_in, d_out, n);
+
+  {
+    const auto actual   = static_cast<std::size_t>(thrust::count(out.begin(), out.end(), 10));
+    const auto expected   = static_cast<std::size_t>(n);
+    REQUIRE(actual==expected);
+  }
+  cudaEventDestroy(event);
+
+}
+
+CUB_TEST("Triple Chevron with missing configuration returns cudaErrorMissingConfiguration", "[test][utils]") {
+  int n = 42;
+  thrust::device_vector<int> in(n, 21);
+  thrust::device_vector<int> out(n);
+  int *d_in  = thrust::raw_pointer_cast(in.data());
+  int *d_out = thrust::raw_pointer_cast(out.data());
+
+  auto chevron = cub::detail::triple_chevron(0, 0);
+  auto err = chevron.doit(mult_two_kernel<int>, d_in, d_out, n);
+  REQUIRE( CubDebug(cudaErrorMissingConfiguration) == cudaErrorMissingConfiguration );
+}
+
+CUB_TEST("Triple Chevron respects required dynamic shared memory allocation", "[test][utils]") {
+  int n = 42;
+  thrust::device_vector<int> in(n, 21);
+  thrust::device_vector<int> out(n);
+  int *d_in  = thrust::raw_pointer_cast(in.data());
+  int *d_out = thrust::raw_pointer_cast(out.data());
+
+  cudaDeviceProp deviceProperties;
+  cudaGetDeviceProperties(&deviceProperties, 0);
+  auto cap = deviceProperties.sharedMemPerBlock;
+
+  const int block_size = 2566666;
+  const int grid_size = (n * block_size - 1) / block_size;
+  auto chevron = cub::detail::triple_chevron(grid_size, block_size, cap); 
+  auto err = chevron.doit(mult_two_kernel<int>, d_in, d_out, n);
+  REQUIRE( CubDebug(cudaErrorInvalidConfiguration) == cudaErrorInvalidConfiguration );
+}
+
+CUB_TEST("Triple Chevron properly forwards parameters", "[test][utils]") {
+  double result;
+  double *d_result;
+
+  cudaMalloc(&d_result, sizeof(double));
+  auto chev = cub::detail::triple_chevron(1, 1); 
+  chev.doit(add_kernel, 5, 3.5f, d_result);
```

Comment:
suggestion: we have different logic for hots and device launch. I'd suggest using cdp helper here as well.